### DISCUSSION TITLE
Welcome to the C language Discussions!

### DISCUSSION THREAD

**Discussion by Panquesito7** [TheAlgorithms/C#810] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together. 💪

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by LoopGlitch26** [comment]
Hello! What're some good places to contribute in C language repository as a starter?

**Comment by Panquesito7** [comment]
I would recommend this repository: mostly all codes have documentation and refined code. You can easily learn from these codes and master your knowledge in C. You can also check out the [Learn to code](https://github.com/collections/learn-to-code) collection at GitHub. 🙂

**Comment by aracnoid149** [comment]
in dictionary.h and dictionary.c, what is the purpose of having MAXELEMENTS set to 1000?

**Reply by rawhuul** [reply]
Look [here](https://github.com/TheAlgorithms/C/blob/d017c3ef77548a74c221cd109f9662a0282d4b23/data_structures/dictionary/dict.h#L23), it's used to declare `elements` which is an array of size `MAXELEMENTS` carrying `void *`. This way one can forget that `elements` is array of some numerical size, rather remember it as array of `MAXELEMENTS` size, you can see [here](https://github.com/TheAlgorithms/C/blob/d017c3ef77548a74c221cd109f9662a0282d4b23/data_structures/dictionary/dict.c#L14), rather than using numerical value, using macro save us in future by just changing value in one place, not everywhere it's used. It's simple technique used more often in projects, it's a pre-processor which replace it's occurence to it's value at compile time.

**Comment by uncodemy14** [comment]
There are many open-source repositories written in C language that you can contribute to as a starter. Here are a few suggestions:

GNU Core Utilities: It's a collection of essential UNIX utilities, such as ls, cp, mv, etc. that are commonly found on Linux and other Unix-like operating systems. You can contribute to this project by fixing bugs, implementing new features, and improving performance.

Libgit2: It's a cross-platform, linkable library for Git that you can use in your own projects. You can contribute to this project by improving documentation, fixing bugs, and adding new features.

SQLite: It's a lightweight, embeddable SQL database engine that is widely used in many applications. You can contribute to this project by improving performance, adding new features, and fixing bugs.

FFmpeg: It's a cross-platform multimedia framework that can encode, decode, transcode, mux, demux, stream, filter, and play audio and video files. You can contribute to this project by improving codecs, adding new filters, and fixing bugs.

Redis: It's an in-memory data structure store that is used as a database, cache, and message broker. You can contribute to this project by improving performance, adding new data structures, and fixing bugs.
